src/pypromice/postprocess/get_bufr.py

```diff
@@ -5,287 +5,343 @@
-		if ('Roof' not in f) and (stid not in to_skip):
-		# if ('v3' not in f) and ('Roof' not in f) and (stid not in to_skip):
-			bufrname = stid + '.bufr'
-			print(f'Generating {bufrname} from {f}')
-
-			if (args.positions is True) and (stid not in positions_update_timestamp_only):
-				positions[stid] = {}
-				# Optionally include source flag columns, useful to indicate if position
-				# comes from current transmission, or older data. This could also be used
-				# to differentiate GPS from modem, but using the combine_first method for
-				# the modem positions currently prevents us from easily knowing which source
-				# was used.
-				# positions[stid]['lat_source'] = ''
-				# positions[stid]['lon_source'] = ''
-
-			# Read csv file
-			df1 = pd.read_csv(f, delimiter=',')
-			df1.set_index(pd.to_datetime(df1['time']), inplace=True)
-			df1.sort_index(inplace=True) # make sure we are time-sorted
-
-			# Check that the last valid index for all instantaneous values match
-			# Note: we cannot always use the single most-recent timestamp in the dataframe
-			# e.g. for 6-hr transmissions, *_u will have hourly data while *_i is nan
-			# Need to check for last valid (non-nan) index instead
-			lvi = {'t_i': df1['t_i'].last_valid_index(),
-				   'p_i': df1['p_i'].last_valid_index(),
-				   'rh_i': df1['rh_i'].last_valid_index(),
-				   'wspd_i': df1['wspd_i'].last_valid_index(),
-				   'wdir_i': df1['wdir_i'].last_valid_index()
-				   }
-
-			two_days_ago = datetime.utcnow() - timedelta(days=2)
-
-			if len(set(lvi.values())) != 1:
-				# instantaneous vars have different timestamps
-				recent = {}
-				for k,v in lvi.items():
-					if (v is not None) and (v >= two_days_ago):
-						recent[k] = v
-				if len(recent) == 0:
-					print('No recent instantaneous timestamps!')
-					no_recent_data.append(stid)
-					if args.positions is True:
-						df1_limited, positions = find_positions(df1, stid, args.time_limit, positions=positions)
-					continue
-				else:
-					# we have partial data, just use the most recent row
-					current_timestamp = max(recent.values())
-					# We will throw this obset down the line, and there is a final min_data_check
-					# to make sure we have minimum data requirements before writing to BUFR
-			else:
-				if all(i is None for i in lvi.values()) is True:
-					print('All instantaneous timestamps are None!')
-					no_valid_data.append(stid)
-					if args.positions is True:
-						df1_limited, positions = find_positions(df1, stid, args.time_limit, positions=positions)
-					continue
-				else:
-					# all values are present, with matching timestamps, so just use t_i
-					current_timestamp = df1['t_i'].last_valid_index()
-
-			print(f'TIMESTAMP: {current_timestamp}')
-
-			# set in dict, will be written to disk at end
-			current_timestamps[stid] = current_timestamp
-
-			if stid in latest_timestamps:
-				latest_timestamp = latest_timestamps[stid]
-
-				if args.dev is True:
```

Comment:
Just want to note, this usage of `args.dev` was really nice. Allows you to run the code repeatedly for testing, without a new transmission. It essentially bypasses the time check.